src/plugins/vis_type_timeseries/server/lib/vis_data/response_processors/series/evaluate.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { create, all } from 'mathjs';
+
+const math = create(all);
+export const evaluate = math.evaluate;
+export const MATH_THROW_MSG = 'Math.js function is disabled';
+
+math.import(
+  {
+    import() {
+      throw new Error(MATH_THROW_MSG);
+    },
+    createUnit() {
+      throw new Error(MATH_THROW_MSG);
+    },
+    evaluate() {
```

Comment:
Is this evaluate() function different than `math.evaluate()` from line 9?

Reply:
It is the same `evaluate` function. The idea is to disable risky functions on the user side. We can find a list of all risky functions in the `mathjs` documentation: https://mathjs.org/docs/expressions/security.html.

So it is possible to do: `mathjs.evaluate('add(2, 2)')`.
At the same time, it is impossible to evaluate inside the evaluate function itself: `mathjs.evaluate('evaluate("add(2,2)")')`.